ivy_tests/test_ivy/test_functional/test_core/test_manipulation.py

```diff
@@ -589,177 +672,96 @@ def test_repeat(
+        split_value = draw(
+            helpers.array_values(
+                dtype=noss_dtype,
+                shape=(1,),
+                min_value=0,
+                max_value=shape[axis] - sum(num_or_size_splits),
             )
         )
-    )
-    _, constant = draw(helpers.dtype_and_values(available_dtypes=[dtype], shape=(1,)))
-    return dtype, value, pad_width, constant[0]
-
-
-# constant_pad
-@settings(deadline=500)
-@given(
-    dtype_value_pad_width_constant=_pad_helper(),
-    as_variable=st.booleans(),
-    with_out=st.booleans(),
-    num_positional_args=helpers.num_positional_args(fn_name="constant_pad"),
-    data=st.data(),
-)
-@handle_cmd_line_args
-def test_constant_pad(
-    *,
-    data,
-    dtype_value_pad_width_constant,
-    as_variable,
-    with_out,
-    num_positional_args,
-    native_array,
-    container,
-    instance_method,
-    fw,
-):
-
-    dtype, value, pad_width, constant = dtype_value_pad_width_constant
+        num_or_size_splits.append(split_value[0])
 
-    helpers.test_function(
-        input_dtypes=dtype,
-        as_variable_flags=as_variable,
-        with_out=with_out,
-        num_positional_args=num_positional_args,
-        native_array_flags=native_array,
-        container_flags=container,
-        instance_method=instance_method,
-        fw=fw,
-        fn_name="constant_pad",
-        x=np.asarray(value, dtype=dtype),
-        pad_width=pad_width,
-        value=constant,
-    )
+    return noss_dtype, num_or_size_splits
 
 
-# zero_pad
-@settings(deadline=500)
 @given(
-    dtype_value_pad_width=_pad_helper(),
-    as_variable=st.booleans(),
-    with_out=st.booleans(),
-    num_positional_args=helpers.num_positional_args(fn_name="zero_pad"),
-    native_array=st.booleans(),
-    container=st.booleans(),
-    instance_method=st.booleans(),
+    noss_type=st.shared(helpers.ints(min_value=1, max_value=2), key="noss_type"),
+    dtype_value=helpers.dtype_and_values(
+        available_dtypes=ivy_np.valid_dtypes,
+        shape=st.shared(helpers.get_shape(min_num_dims=1), key="value_shape"),
+    ),
+    axis=st.shared(
+        helpers.get_axis(
+            shape=st.shared(helpers.get_shape(min_num_dims=1), key="value_shape"),
+            force_int=True,
+        ),
+        key="target_axis",
+    ),
+    with_remainder=st.booleans(),
+    num_or_size_splits=_split_helper(),
+    num_positional_args=helpers.num_positional_args(fn_name="split"),
     data=st.data(),
 )
 @handle_cmd_line_args
-def test_zero_pad(
+def test_split(
     *,
     data,
-    dtype_value_pad_width,
+    noss_type,
+    dtype_value,
+    num_or_size_splits,
+    axis,
+    with_remainder,
     as_variable,
-    with_out,
     num_positional_args,
     native_array,
     container,
     instance_method,
     fw,
 ):
 
-    # Drop the generated constant as only 0 is used
-    dtype, value, pad_width, _ = dtype_value_pad_width
+    dtype, value = dtype_value
+
+    if noss_type == 2:
```

Comment:
add comment

Reply:
Code optimisations removed this if statement